plugins/modules/proxmox.py

```diff
@@ -733,8 +798,43 @@ def main():
     # Create a new container
     if state == 'present' and clone is None:
         try:
-            if proxmox.get_vm(vmid, ignore_missing=True) and not module.params['force']:
-                module.exit_json(changed=False, vmid=vmid, msg="VM with vmid = %s is already exists" % vmid)
+            if proxmox.get_vm(vmid, ignore_missing=True):
+                if module.params["update"]:
+                    try:
+                        proxmox.update_config(vmid, node, disk, cpus, memory, swap,
+                                              cores=module.params["cores"],
+                                              hostname=module.params["hostname"],
+                                              netif=module.params["netif"],
+                                              mounts=module.params["mounts"],
+                                              ip_address=module.params["ip_address"],
+                                              onboot=ansible_to_proxmox_bool(module.params["onboot"]),
+                                              cpuunits=module.params["cpuunits"],
+                                              nameserver=module.params["nameserver"],
+                                              searchdomain=module.params["searchdomain"],
+                                              features=",".join(module.params["features"])
+                                              if module.params["features"] is not None
+                                              else None,
+                                              description=module.params["description"],
+                                              hookscript=module.params["hookscript"],
+                                              timezone=module.params["timezone"],
+                                              tags=module.params["tags"])
+                        module.exit_json(
+                            changed=True,
```

Comment:
Is it possible to detect whether there are no changes? If yes, that should be done. If not, the documentation should be very clear that the module is not able to detect this.

Reply:
Good catch, I can try fetching the current config from the API and diffing against the new requested config.

Reply:
BTW, the proxmox_kvm module also has an `update` parameter, and it seems to have the same problem - see #1965. https://github.com/ansible-collections/community.general/issues/1965#issuecomment-1043439573 might be relevant here as well.

Reply:
I'm comparing against the current running config now. I tested against the majority of config options and everything seemed good, but didn't test every option.